src/components/ui/form/switch/Switch.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { ChangeEvent, useCallback } from 'react';
+import styled, { css } from 'react-emotion';
+import { compose, setDisplayName } from 'recompose';
+import { RequiredInputComponentProps } from '../../../RequiredComponentProps';
+import {
+  withComponentTheme,
+  WithInnerComponentThemeProps,
+} from '../../../util/enhancers';
+import { Icon } from '../../icon';
+import { SwitchTheme } from './SwitchTheme';
+
+export interface SwitchProps
+  extends RequiredInputComponentProps<HTMLInputElement> {
+  checked: boolean;
+  disabled?: boolean;
+  onChange: (value: boolean) => void;
+  value: string;
+  theme?: SwitchTheme;
+}
+
+const invisibleInputStyles = css`
+  top: 0;
+  left: 0;
+  width: 100%;
+  cursor: inherit;
+  height: 100%;
+  margin: 0;
+  opacity: 0;
+  padding: 0;
+  position: absolute;
+`;
+
+const Back = styled('div')<Pick<SwitchProps, 'checked' | 'disabled' | 'theme'>>`
+  cursor: ${({ disabled }) => (disabled ? 'inherit' : 'pointer')};
+  width: ${({ theme }) => theme.width}px;
+  height: ${({ theme }) => theme.height}px;
+  border-radius: ${({ theme }) => theme.borderRadius}px;
+  background-color: ${({ checked, disabled, theme }) =>
+    disabled
+      ? theme.disabledColors.backgroundColor
+      : checked
+        ? theme.checkedColors.backgroundColor
+        : theme.colors.backgroundColor};
+  position: relative;
+
+  :hover {
+    opacity: ${({ disabled }) => (disabled ? '1' : '0.7')};
+  }
+`;
+
+const Front = styled('div')<
+  Pick<SwitchProps, 'checked' | 'disabled' | 'theme'>
+>`
+  background-color: ${({ checked, disabled, theme }) =>
+    disabled
+      ? theme.disabledColors.iconBackgroundColor
+      : checked
+        ? theme.checkedColors.iconBackgroundColor
+        : theme.colors.iconBackgroundColor};
+  border-radius: ${({ theme }) => theme.borderRadius - 1}px;
+  height: ${({ theme }) => theme.height - 4}px;
+  position: absolute;
+  right ${({ checked, theme }) =>
+    checked ? 2 : `${theme.width - theme.height + 2}`}px;
+  top: 2px;
+  transition: right ${({ theme }) => theme.width / 400}s linear;
```

Comment:
We should have standard transition times, that everyone uses.

I usually have something like

```
const TransitionSpeed = {
   SLOW: '500ms',
   MEDIUM: '250ms',
   FAST: '100ms',
}
```

It probably shouldn't depend on the width. If the width is 400px, the animation will take 1s which is really slow :)

Reply:
Also, you should probably change to an easing animation, instead of `linear`.
Maybe `ease-out`?